interfaces/Portal/src/app/models/user.model.ts

```diff
@@ -1,14 +1,7 @@
-import Permission from '../auth/permission.enum';
+import { UserController } from '@121-service/src/user/user.controller';
+import { Dto121Service } from '../shared/utils/dto-type';
 
-export class User {
-  username: string;
-  permissions: {
-    [programId: number]: Permission[];
-  };
-  expires?: string;
-  isAdmin?: boolean;
-  isEntraUser?: boolean;
-}
+export type User = Dto121Service<UserController['login']>['user'];
```

Comment:
This is the main change, and the one that would need to be applied in the rest of the codebase.

As such, I would not recommend actually doing it in `Portal`, and to simply start `Portalicious` with this approach instead.

Reply:
Ah, so it would have a similar 'workaround' like we're already using the `PermissionsEnum`-type from the 121-service, with a sort-of "local proxy"-file...

So components/services in the Portal(icious) would only have to worry about importing this local model/type-file. And only in those "proxy"-files, we would have the "import from `@121-service`-magic..

Reply:
Exactamundo 👌